server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -2689,7 +2691,19 @@ public BlockLoader blockLoader(MappedFieldType.BlockLoaderContext blContext) {
             }
 
             if (indexed) {
-                return new BlockDocValuesReader.DenseVectorBlockLoader(name(), dims, this);
+                BlockLoaderFunction<?> blockLoaderFunction = blContext.blockLoaderFunction();
+                if (blockLoaderFunction == null) {
+                    return new BlockDocValuesReader.DenseVectorBlockLoader(name(), dims, this);
+                }
+                if (SIMILARITY_FUNCTION_NAME.equals(blockLoaderFunction.name()) == false) {
```

Comment:
I'm not super fan of using a name to identify the function. I think we could get away with checking the type of the config 🤔

Reply:
Yeah. I think you can just:
```
if (blockLoaderFunction == null) {
                    return new BlockDocValuesReader.DenseVectorBlockLoader(name(), dims, this);
}
if (blockLoaderFunction instanceof DenseVectorSimilarityFunctionBlockLoader config) {
                  return new BlockDocValuesReader.DenseVectorSimilarityFunctionBlockLoader(
}
                    throw new UnsupportedOperationException("Unknown block loader function: " + blockLoaderFunction);
```

The only thing is - we need to make sure there's a nice toString for the config.